packages/babel-helper-module-transforms/src/normalize-and-load-metadata.js

```diff
@@ -371,16 +374,19 @@ function getLocalExportMetadata(
           if (exported.node.name === "__esModule") {
             throw exported.buildCodeFrameError('Illegal export "__esModule".');
           }
-          getLocalMetadata(local).names.push(exported.node.name);
+          getLocalMetadata(local).addName(exported.node.name);
         });
       }
     } else if (child.isExportDefaultDeclaration()) {
       const declaration = child.get("declaration");
-      if (
+
+      if (declaration.isIdentifier()) {
```

Comment:
Aren't identifiers normal expressions? e.g., what is the difference between
```js
var a = 2;
export default a;
```
and
```js
export default 2;
```
?

Reply:
There is no difference when it comes to the functionality.

The containing function here (`getLocalExportMetadata`) gathers export names for local bindings so every future assignment to those local bindings can be transformed into a live reference.

You can note that this function here expects only a FunctionDeclaration, ClassDeclarations and now Identifiers in position of default export. Everything else in this position gets splitted before by `nameAnonymousExports` into
```
var *local* = expression
export { *local* as default }
```
I omit this splitting for local bindings [here](https://github.com/babel/babel/blob/cb4eface89b9837c0ac02f9e3473263b07759e91/packages/babel-helper-module-transforms/src/normalize-and-load-metadata.js#L412-L417) to produce a nicer output, if I'd keep splitting them we would end up with things like:
```js
var _default = function () { /* some class */ }
var _default2 = _default
module.exports = _default2
```